yt_dlp/extractor/zingmp3.py

```diff
@@ -67,6 +66,20 @@ def _parse_items(self, items):
         for url in traverse_obj(items, (..., 'link')) or []:
             yield self.url_result(urljoin(self._DOMAIN, url))
 
+    def _fetch_page(self, *arg):
+        raise NotImplementedError('This method must be implemented by subclasses')
+
+    def _paged_list(self, _id, url_type):
+        page = 1
+        entries = []
+        while True:
+            data = self._fetch_page(_id, url_type, page)
+            entries.extend(self._parse_items(data.get('items')))
+            if not data.get('hasMore') or len(entries) > data.get('total'):
+                break
+            page += 1
+        return entries
```

Comment:
Consider using a PagedList

Reply:
Current version used PagedList. But it seem like not good for this case. Because Zingmp3 use hasMore key to handle pagination when scroll.

Reply:
You r right. My bad

Reply:
Wouldn't it work if you just remove the check and make it a pagedlist?